dump_zenodo_open_metadata.py

```diff
@@ -53,53 +53,65 @@ def parse_record(rec, spam=False):
     out["spam"] = spam
     return out
 
+print("What type of dump you want? 1- Ham, 2- Spam or 3- Both?")
+type = input()
+
+if(type == 1):
+    type_string = "ham_"
+elif(type == 2):
+    type_string = "spam_"
+else:
+    type_string = "ham&spam_"
+
 today = datetime.date.today().isoformat()
-FILENAME = f"zenodo_open_metadata_{today}.jsonl.gz"
+FILENAME = "zenodo_open_metadata_" + type_string + today + ".jsonl.gz"
 
-records_ham = (
-    db.session.query(RecordMetadata)
-    .join(PersistentIdentifier, PersistentIdentifier.object_uuid == RecordMetadata.id)
-    .filter(
-        PersistentIdentifier.pid_type == "recid",
-        PersistentIdentifier.status == "R",
-        PersistentIdentifier.object_type == "rec",
+if(type != 2):
+    records_ham = (
+        db.session.query(RecordMetadata)
+        .join(PersistentIdentifier, PersistentIdentifier.object_uuid == RecordMetadata.id)
+        .filter(
+            PersistentIdentifier.pid_type == "recid",
+            PersistentIdentifier.status == "R",
+            PersistentIdentifier.object_type == "rec",
+        )
     )
-)
 
-# Write the non-spam records first
-total_ham = records_ham.count()
-with gzip.open(FILENAME, "wb") as fp:
-    for idx, r in enumerate(records_ham.yield_per(1000)):
-        if idx % 10000 == 0:
-            print('[{}]: {} / {}'.format(datetime.datetime.now().isoformat(), idx, total_ham))
-        if r.json is not None:
-            rec = parse_record(Record(r.json, model=r), spam=False)
-            fp.write(json.dumps(rec) + "\n")
+    # Write the non-spam records first
+    total_ham = records_ham.count()
+    with gzip.open(FILENAME, "wb") as fp:
```

Comment:
```suggestion
    with gzip.open(FILENAME, "ab") as fp:
```

Since we're now always "adding" to the file (in case one has only chosen ham/spam), we always use the `ab` option (to append-bytes)